Add contribution guidelines to the documentation (#843)
ROCm/composable_kernel: docs/Contributors_Guide.rst

issue. All reported issues must include:

* A comprehensive description of the problem, including:

  * What did you observe?
  * Why do you think it is a bug (if it seems like one)?
  * What did you expect to happen? What would indicate the resolution of the problem?
  * Are there any known workarounds?

* Your configuration details, including:

  * Which GPU are you using?
  * Which OS version are you on?
  * Which ROCm version are you using?
  * Are you using a Docker image? If so, which one?

* Steps to reproduce the issue, including:

  * What actions trigger the issue? What are the reproduction steps?

    * If you build the library from scratch, what CMake command did you use?

  * How frequently does this issue happen? Does it reproduce every time? Or is it a sporadic issue?

Before sumbitting any issue, ensure you have addressed all relevant questions from the checklist.

Creating Pull Requests
----------------------

You can submit `Pull Requests (PR) on GitHub
<https://github.com/ROCmSoftwarePlatform/composable_kernel/pulls>`_.

All contributors are required to develop their changes on a separate branch and then create a

> pull requrest to merge their changes into the `develop` branch, which is the default

Reviewer: Isn't like users outside of ROCmSoftwarePlatform have to create their own fork of CK and can create PR only from their fork to our repo ?

Author: right, I added another sentence about the need for using forks

development branch in the Composable Kernel project.

When submitting a Pull Request you should:

* Describe the change providing information about the motivation for the change and a general
  description of all code modifications.

* Verify and test the change:

  * Run any relevant existing tests.

>   * Write new tests if the existing ones do not exist or do not cover the functionality entirely.

Reviewer: ```suggestion
  * Write new tests if added functionality is not covered by current tests.
```

* Link your PR to any related issues:

  * If there is an issue that is resolved by your change, please provide a link to the issue in
    the description of your pull request.

>   * If there is no issue related to your change, we recommend creating a new issue and then

Reviewer: This point seems to me ambiguous. Not every PR is fixing some issues.

Author: removed
